stateproof/README.md

```diff
@@ -0,0 +1,39 @@
+# StateProof
+
+The stateproof package implements the orchestration logic for state proof. 
+While the crypto package implements the cryptography behind state proofs, this package is responsible for
+- Signatures collection
+- Signatures production
+- Ledger data gathering
+- State proof transaction generation.
+
+
+## State Proof Modules
+
+- `verify` responsible for making a decision on whether or not to accept a state proof with respect to the round
+  in which it is was proposed. The network aims to accept the most compact state proof
+  it can produce while on the other hand, produce a proof every `StateProofInterval` rounds.
+  For that reason, the network might not accept a state proof even if it is valid in order to
```

Comment:
```suggestion
  For this reason, the network might not accept a valid State Proof when there is a chance a better (more compact) State Proof could be produced. 
```
This is a really interesting detail. How does the network decide whether or not to accept a valid State Proof, then?